### PR TITLE
Switch to using Pulumi components

### PR DESCRIPTION
**Pull request by s-block**
Initial Pulumi for the DE Buckets, Users and VPN

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -1,13 +1,45 @@
...
+
+
+/**
+ * Data Buckets
+ */
+const historyBucketName = config.require("de:historyBucketName");
+const rawBucket = new DataBucket(historyBucketName, {
+    logBucketId: logBucket.id
+});
+export const rawBucketId = rawBucket.bucket.id;
```
Comment (xoen) [inline index.ts]: are these the `historyBucket` and the `historyBucketID`? or are these unrelated to the history bucket (whatever that is)

File: index.ts [hunk 2]
```diff
@@ -1,13 +1,45 @@
...
+
+const deLogBucketName = config.require("de:deLogBucketName");
```
Comment (xoen) [inline index.ts]: just out of curiosity, what does `de` stands for? de-duplication or something like that?

File: index.ts (hunk 1 above)
Comment (s-block) [reply index.ts]: Yeah, same thing. Probably should be `rawHistoryBucket` I will rename

File: index.ts [hunk 3]
```diff
@@ -17,7 +49,76 @@ bucketPolicy.allowIamToPutToPath(
         "arn:aws:iam::649098267436:role/OpsDelegatee"
     ]
 );
-const bucket = createLandingBucket("mojap-land", bucketPolicy, logBucket.id);
+const landingBucketName = config.require("de:deLogBucketName");
```
Comment (xoen) [inline index.ts]: Is this supposed to be `= config.require("de:deLogBucketName");`? I'm asking as constant is called `landingBucketName`.

File: index.ts (hunk 2 above)
Comment (s-block) [reply index.ts]: data engineering - the config needs to be namespaces with something

File: index.ts (hunk 3 above)
Comment (s-block) [reply index.ts]: Yeah correct. I have changed the key

File: index.ts (hunk 2 above)
Comment (xoen) [reply index.ts]: oh yes it didn't occur to me few minutes ago this was short for Data Engineering 🤦‍♂

**Pull request merged**